pump/server.go

```diff
@@ -399,14 +409,16 @@ func (s *Server) Start() error {
 
 	log.Info("register success", zap.String("NodeID", s.node.NodeStatus().NodeID))
 
-	// notify all cisterns
-	ctx, _ := context.WithTimeout(s.ctx, notifyDrainerTimeout)
-	if err := s.node.Notify(ctx); err != nil {
-		// if fail, refresh this node's state to paused
-		if err := s.registerNode(context.Background(), node.Paused, 0); err != nil {
-			log.Error("unregister pump while pump fails to notify drainer", zap.Error(err))
+	// notify all cisterns when this pump node is a newly registered one
+	if previousStatus == nil || previousStatus.State != node.Paused {
```

Comment:
!= node.Offline

Reply:
You mean change `node.Paused` to `node.Offline`?

Reply:
yes

Reply:
For nodes whose previous status is `Offline` collector doesn't have a record in `collector.pumps`. I think we should notify the drainer in this condition. Current version means this node hasn't registered before or it isn't starting from `Paused` status and in these conditions we should notify `collector`.